articles/introduction-to-friend-functions-in-c++/index.md

```diff
@@ -0,0 +1,202 @@
+
+Example of a program to illustrate friend class:
+```c++
+//program to illustrate
+//friend class
+#include <iostream>
+using namespace std;
+
+// forward declaration
+class ClassY;
+
+class ClassX {
+        int digit1;
+
+        // friend class declaration
+        friend class ClassY;
+
+    public:
+        // constructor to initialize num1 to 10
+        ClassX() : digit1(10) {}
+};
+
+class ClassY {
+        int digit2;
+
+    public:
+        // constructor to initialize num2 to 5
+        ClassY() : digit2(5) {}
+
+    // member function to multiply num1
+    // from ClassX with num2 from ClassY
+    int multiply() {
+        ClassX m;
+        return m.digit1 * digit2;
+    }
+};
+
+int main() {
+    ClassY n;
+    cout << "Multiplication: " << n.multiply();
+    return 0;
+}
+```
+The output will be:
+`50`
+
+In the program above, we have declared two classes X and Y. `ClassY` is a friend class of `ClassX`. Therefore, `ClassY` has access to the member function of `classX`. In ClassY, we have created a function` multiply()` that returns the multiplication of `digit1` and `digit2`. `ClassY` being a friend class enables us to create objects of `ClassX` inside of ClassY. This is possible the through forward declaration of the `classY`. 
+
+#### Here are some important points to note on friend functions and classes: 
+- Friend functions should be used for restricted purposes only. Because having excessive friend functions and classes can reduce the Object-oriented programming feature of encapsulation in a program.
+- Friendship is not reciprocal. If class X is Y’s friend, then Y does not automatically become X’s friend.
+- Friendship cannot be inherited.
+
+### Conclusion
+In this article, we got to explore friend functions and friend class, what they and where they are used. By making a function and class "friendly" we reduce having too many functions therefore, we have programs that are maintainable and easy to read.
+
+### Additional Resources
+Dawe, D. Introduction to friend functions in C++. Retrieved from https://codevidyalay.blogspot.com/2019/09/friend-class-and-friend-function-in-c.html.
+
+Dawe, D. Introduction to friend functions in C++. Retrieved from https://www.indiabix.com/cpp-programming/objects-and-classes/discussion-170.
+
+Dawe, D. Introduction to friend functions in C++. Retrieved from https://mounanaravani.files.wordpress.com/2017/02/17-friend-functions.pdf.
+
+Dawe, D. Introduction to friend functions in C++. Retrieved from https://www.softwaretestinghelp.com/friend-functions-in-cpp.
+
+Dawe, D. Introduction to friend functions in C++. Retrieved from http://web.archive.org/web/20191023010010/http://lessons2all.com:80/ff.php.
+
+Dawe, D. Introduction to friend functions in C++. Retrieved from https://www.iditect.com/how-to/10227802.html.
+
+Dawe, D. Introduction to friend functions in C++. Retrieved from https://www.programiz.com/cpp-programming/friend-function-class.
```

Comment:
Can you add the resource title instead of `Dawe, D. Introduction to friend functions in C++.`